### ISSUE TITLE
Haskell highlighting is broken

### ISSUE THREAD

**Issue by phaazon** [morhetz/gruvbox#190]
I had to remove the overridings from gruvbox to get a decent syntax highlighting back. There’s some kind of regression. Is there a way to fix that?

**Comment by morhetz** [comment]
elaborate please

**Comment by phaazon** [comment]
Yeah, sorry. I’ll try to provide a screenshot different when I meet enough spare time. The issue is that, now, for instance, the signature functions are like plain text, they don’t even have colors any more. And it’s the same for most common Haskell constructs, like operators.

**Comment by Adaephon-GH** [comment]
I ran into the same issue and did some digging. It seems that the cause for the broken (or only partially working) highlighting is that Gruvbox does not link to the default highlighting group names.

In the haskell syntax definition that is included with `vim` and `nvim` (`$VIMRUNTIME/syntax/haskell`) every group name starts with `hs` followed by the appropriate specifier. For example `hsString`, `hsCharacter`, `hsOperator`. Whereas the groups linked to by Gruvbox have the prefix `haskell`, for example `haskellString`, `haskellChar`, `haskellOperators`.

It seems to me that, when writing the highlighting for Haskell, some other plugin was active that changed the syntax definition for Haskell. One candidate might be [haskell-vim](https://github.com/neovimhaskell/haskell-vim) as using it indeed improves highlighting with Gruvbox.

**Comment by morhetz** [comment]
@Adaephon-GH https://github.com/sheerun/vim-polyglot is the base syn package for modern vim
so, yeah it's https://github.com/neovimhaskell/haskell-vim

**Comment by phaazon** [comment]
Is there any leads to fix that problem? I might switch to a fork until it gets fixed (why overriding the syntax hilighting in the first place?)

**Comment by KevinSjoberg** [comment]
@morhetz any update here?

**Comment by mitchellwrosen** [comment]
I agree syntax highlighting looks a bit bland in Haskell, but I'm not sure I understand the explanation above, nor the fix merged into `gruvbox-community`.

Looking at the syntax definition in gruvbox, both `haskellIdentifier` and `haskellType` are defined as `GruvboxFg1`. The result is a lot of white where perhaps some more color would look nicer. Here's a random example screenshot:
![2020-09-08-100916_1919x1059_scrot](https://user-images.githubusercontent.com/1074598/92487565-62fabf80-f1bb-11ea-8056-600edc8cff7f.png)
